CSS calc

Hi!

I got some weird results with `calc`.

## Version
1.0.11

## Steps to reproduce

```javascript
const DoubleInputContainer = styled.div`
  margin-bottom: calc(15px - 0.5rem) !important;
`
```

## Expected Behavior

```css
margin-bottom: calc(15px - 0.5rem) !important;
```

## Actual Behavior (Chrome console)

```css
margin-bottom: calc(15px - 0.5rem); /* !important is missing */
marginBottom: -webkit-calc(15px - 0.5rem) !important; /* marginBottom is not valid */
```

Thanks for the help!

Could this change of autoprefixer in **glamor** be the reason I get: 

`Error in ./~/styled-components/lib/utils/autoprefix.js`
`Module not found: Error: Cannot resolve module 'glamor/lib/autoprefix' in [...]/node_modules/styled-components/lib/utils`
`@ ./~/styled-components/lib/utils/autoprefix.js 15:18-50`

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -12,7 +12,7 @@ All notable changes to this project will be documented in this file. If a contri
 
 ### Changed
 
-- N/A
+- Switched to using [inline-style-prefixer](https://github.com/rofrischmann/inline-style-prefixer) for our autoprefixing needs.
 
 ## [v1.0.11] - 2016-11-14
 

--- a/package.json
+++ b/package.json
@@ -40,6 +40,7 @@
     "buffer": "^5.0.0",
     "fbjs": "^0.8.4",
     "glamor": "^2.15.5",
+    "inline-style-prefixer": "^2.0.4",
     "lodash": "^4.15.0",
     "supports-color": "^3.1.2"
   },

--- a/src/test/css.test.js
+++ b/src/test/css.test.js
@@ -5,7 +5,7 @@ import { resetStyled, expectCSSMatches } from './utils'
 
 let styled
 
-describe('css features', () => {
+describe.only('css features', () => {
   beforeEach(() => {
     styled = resetStyled()
   })
@@ -16,6 +16,20 @@ describe('css features', () => {
     `
     shallow(<Comp />)
     expectCSSMatches('.a { -ms-transition: opacity 0.3s; -moz-transition: opacity 0.3s; -webkit-transition: opacity 0.3s; transition: opacity 0.3s; }')
+  })
+
+  it('should handle CSS calc()', () => {
+    const Comp = styled.div`
+      margin-bottom: calc(15px - 0.5rem) !important;
+    `
+    shallow(<Comp />)
+    expectCSSMatches(`
+      .a { 
+        margin-bottom: calc(15px - 0.5rem) !important;
+        marginBottom:-webkit-calc(15px - 0.5rem) !important;
+        marginBottom:-moz-calc(15px - 0.5rem) !important;
+      }
+    `)
   })
 
   it('should pass through custom properties', () => {

--- a/src/test/css.test.js
+++ b/src/test/css.test.js
@@ -5,7 +5,7 @@ import { resetStyled, expectCSSMatches } from './utils'
 
 let styled
 
-describe.only('css features', () => {
+describe('css features', () => {
   beforeEach(() => {
     styled = resetStyled()
   })
@@ -15,7 +15,34 @@ describe.only('css features', () => {
       transition: opacity 0.3s;
     `
     shallow(<Comp />)
-    expectCSSMatches('.a { -ms-transition: opacity 0.3s; -moz-transition: opacity 0.3s; -webkit-transition: opacity 0.3s; transition: opacity 0.3s; }')
+    expectCSSMatches('.a { -webkit-transition: opacity 0.3s; transition: opacity 0.3s; }')
+  })
+
+  it('should add vendor prefixes for display', () => {
+    const Comp = styled.div`
+      display: flex;
+      flex-direction: column;
+      align-items: center;
+    `
+    shallow(<Comp />)
+    expectCSSMatches(`
+      .a {
+        display: -webkit-box;
+        display: -moz-box;
+        display: -ms-flexbox;
+        display: -webkit-flex;
+        display: flex;
+        -webkit-box-direction: normal;
+        -webkit-box-orient: vertical;
+        -ms-flex-direction: column;
+        -webkit-flex-direction: column;
+        flex-direction: column;
+        -webkit-box-align: center;
+        -ms-flex-align: center;
+        -webkit-align-items: center;
+        align-items: center;
+      }
+    `)
   })
 
   it('should handle CSS calc()', () => {
@@ -24,10 +51,10 @@ describe.only('css features', () => {
     `
     shallow(<Comp />)
     expectCSSMatches(`
-      .a { 
+      .a {
+        margin-bottom: -webkit-calc(15px - 0.5rem) !important;
+        margin-bottom: -moz-calc(15px - 0.5rem) !important;
         margin-bottom: calc(15px - 0.5rem) !important;
-        marginBottom:-webkit-calc(15px - 0.5rem) !important;
-        marginBottom:-moz-calc(15px - 0.5rem) !important;
       }
     `)
   })

--- a/src/utils/autoprefix.js
+++ b/src/utils/autoprefix.js
@@ -2,10 +2,8 @@
 import camelizeStyleName from 'fbjs/lib/camelizeStyleName'
 import hyphenateStyleName from 'fbjs/lib/hyphenateStyleName'
 
-// eslint-disable-next-line import/no-duplicates
-import { autoprefix } from 'glamor/lib/autoprefix'
-// eslint-disable-next-line import/no-duplicates
-import typeof { Container } from 'glamor/lib/autoprefix'
+import prefixAll from 'inline-style-prefixer/static'
+import type Container from '../vendor/postcss/container'
 
 export default (root: Container) => {
   root.walkDecls(decl => {
@@ -13,11 +11,15 @@ export default (root: Container) => {
     if (decl.prop.startsWith('--')) return
 
     const objStyle = { [camelizeStyleName(decl.prop)]: decl.value }
-    const prefixed = autoprefix(objStyle)
+    const prefixed = prefixAll(objStyle)
     Object.keys(prefixed).reverse().forEach(newProp => {
-      decl.cloneBefore({
-        prop: hyphenateStyleName(newProp),
-        value: prefixed[newProp],
+      const newVals = prefixed[newProp]
+      const newValArray = Array.isArray(newVals) ? newVals : [newVals]
+      newValArray.forEach(newVal => {
+        decl.cloneBefore({
+          prop: hyphenateStyleName(newProp),
+          value: newVal,
+        })
       })
     })
     decl.remove()

